File: l5_data/user_db.py
# l5_data/user_db.py

class UserDB:
    def __init__(self):
        # 模拟数据库存储
        self.transactions = []
        self.holdings = []

    def get_user_holdings(self, user_id):
        """
        获取用户持仓
        Mock implementation. Real implementation would query a DB.
        """
        # 模拟返回一些持仓数据
        return [
            {"ticker": "0700.HK", "cost": 300.0, "quantity": 100},
            {"ticker": "BABA", "cost": 80.0, "quantity": 50},
            {"ticker": "AAPL", "cost": 150.0, "quantity": 20}
        ]

    def get_recent_transactions(self, user_id, limit=5):
        """
        获取最近交易流水
        """
        return [
            {"date": "2023-10-01", "action": "BUY", "ticker": "0700.HK", "price": 300.0, "quantity": 100},
            {"date": "2023-09-15", "action": "BUY", "ticker": "BABA", "price": 80.0, "quantity": 50}
        ]

File: src/l5_data/user_db.ts
import { Holding, Transaction, TradingBehavior } from '../types';

/**
 * L5 Data - User Database
 * 用户数据库，管理用户持仓和交易记录
 */
export class UserDB {
  private transactions: Transaction[] = [];
  private holdings: Holding[] = [];

  /**
   * 获取用户持仓
   * @param userId 用户ID
   * @returns 持仓列表
   */
  getUserHoldings(userId: string): Holding[] {
    // 模拟返回一些持仓数据
    return [
      { ticker: '0700.HK', cost: 300.0, quantity: 100 },
      { ticker: 'BABA', cost: 80.0, quantity: 50 },
      { ticker: 'AAPL', cost: 150.0, quantity: 20 },
    ];
  }

  /**
   * 获取最近交易流水
   * @param userId 用户ID
   * @param limit 返回记录数量限制
   * @returns 交易记录列表
   */
  getRecentTransactions(userId: string, limit: number = 5): Transaction[] {
    return [
      {
        date: '2023-10-01',
        action: 'BUY',
        ticker: '0700.HK',
        price: 300.0,
        quantity: 100,
      },
      {
        date: '2023-09-15',
        action: 'BUY',
        ticker: 'BABA',
        price: 80.0,
        quantity: 50,
      },
    ];
  }
}
